File: Arrays/1266. Minimum Time Visiting All Points/1266. Minimum Time Visiting All Points.py
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        time = 0
        for i in range(len(points) - 1):
            current = points[i]
            nextt = points[i + 1]
            x = abs(nextt[0] - current[0])
            y = abs(nextt[1] - current[1])
            time += max(x, y)
        
        return time

# Time: O(n)
# Space: O(1)

File: Arrays/1266. Minimum Time Visiting All Points/1266. Minimum Time Visiting All Points.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minTimeToVisitAllPoints = function (points) {
  let time = 0;
  for (let i = 0; i < points.length - 1; i++) {
    current = points[i];
    next = points[i + 1];
    x = Math.abs(next[0] - current[0]);
    y = Math.abs(next[1] - current[1]);
    time += Math.max(x, y);
  }
  return time;
};

// Time: O(n)
// Space: O(1)
